Bumbodium.Data/CaoInput.cs

```diff
@@ -0,0 +1,152 @@
+                results.Add(MaxAverageHoursAWeek(40));
+            }
+
+            //under 16
+            if (_employee.Age < 16)
+            {
+                results.Add(MaxShiftsThisWeek(5));
+                results.Add(MaxAmountHourShift(8));
+                results.Add(LatestTime(new TimeOnly(19, 0)));
+
+                foreach(var vacationWeek in _vacationWeeks)
+                {
+                    if (_plannedShift.ShiftStartDateTime.DayOfYear / 7 == vacationWeek)
+                        results.Add(MaxHoursAWeek(40));
+                    else
+                        results.Add(MaxHoursAWeek(12));
+                }
+            }
+
+            return results;
+        }
+
+        //Calculates the break in minutes
+        public TimeSpan CalcBreak()
+        {
+            TimeSpan breaks = new TimeSpan();
+            int workingHours = _plannedShift.ShiftEndDateTime.Hour - _plannedShift.ShiftStartDateTime.Hour;
+            int fullBreaks = (workingHours / 8) + (int)(workingHours / 4.5);
+            for(int i = 0; i < fullBreaks; i++)
+            {
+                breaks.Add(new TimeSpan(0, 30, 0));
+            }
+
+            return breaks;
+        }
+
+        #region IndividualRules
+
+        //Returns validation result if the planned shift is longer than the max amount given
+        private CaoValidationResult? MaxAmountHourShift(int maxHours)
+        {
+            int schoolHours = 0;
+            if (_employee.Availability != null)
+            {
+                var schoolAvailabilities = _employee.Availability.Where(a => a.Type == AvailabilityType.Schoolhours &&
+                                                           a.StartDateTime.Day == _plannedShift.ShiftStartDateTime.Day);
+                foreach (var availability in schoolAvailabilities)
+                    schoolHours += availability.EndDateTime.Hour - availability.StartDateTime.Hour;
+            }
+            
+            if (_plannedShift.ShiftStartDateTime.Hour - _plannedShift.ShiftEndDateTime.Hour > (maxHours - schoolHours))
+                return new CaoValidationResult($"Deze werknemer mag maximaal maar {maxHours} uur in één dienst werken", "MaxAmountHourShift");
+            return null;
+        }
+
+        //Returns validation result if the average over a month is more than the max amount given
+        private CaoValidationResult? MaxAverageHoursAWeek(int maxHours)
+        {
+            int hoursThisMonth = 0;
+            var month = _plannedShift.ShiftStartDateTime.Month;
+            foreach(var shift in _workedShifts)
+            {
+                if(shift.ShiftStartDateTime.Month == month)
+                    hoursThisMonth += shift.ShiftEndDateTime.Hour - shift.ShiftStartDateTime.Hour;
+            }
+            if (hoursThisMonth / 7 > maxHours)
+                 return new CaoValidationResult($"Deze werknemer mag gemiddeld maar {maxHours} uur per week per maand werken", "MaxAverageHoursAWeek");
```

Comment:
Per week per maand?

Reply:
gemiddeld per week, per maand. Dus bijv:
week 1: 50 uur
week 2: 30 uur
week 3: 40 uur
week 4: 40 uur
is gemiddeld 40 uur per week, over één maand